__tests__/distributors/zulu.test.ts

```diff
@@ -0,0 +1,525 @@
+import fs from 'fs';
+
+import * as tc from '@actions/tool-cache';
+import * as core from '@actions/core';
+import * as ifm from '@actions/http-client/interfaces';
+import { HttpClient } from '@actions/http-client';
+
+import path from 'path';
+import * as semver from 'semver';
+
+import * as utils from '../../src/util';
+import { ZuluDistributor } from '../../src/distributors/zulu/installer';
+import { IZuluVersions } from '../../src/distributors/zulu/models';
+
+let manifestData = require('../data/zulu/zulu-releases-default.json');
+
+describe('getAvailableVersions', () => {
+  let spyHttpClient: jest.SpyInstance;
+  let zuluDistributor: ZuluDistributor;
+  let originalPlatform: NodeJS.Platform;
+
+  beforeEach(() => {
+    spyHttpClient = jest.spyOn(HttpClient.prototype, 'getJson');
+    spyHttpClient.mockImplementation(
+      async (): Promise<ifm.ITypedResponse<IZuluVersions[]>> => {
+        const result = JSON.stringify(manifestData);
+        return {
+          statusCode: 200,
+          headers: {},
+          result: JSON.parse(result) as IZuluVersions[]
+        };
+      }
+    );
+
+    originalPlatform = process.platform;
+    Object.defineProperty(process, 'platform', {
+      value: 'darwin'
+    });
+  });
+
+  afterEach(() => {
+    Object.defineProperty(process, 'platform', {
+      value: originalPlatform
+    });
+    jest.resetAllMocks();
+    jest.clearAllMocks();
+    jest.restoreAllMocks();
+  });
+
+  it.each([
+    [
+      { version: '11', arch: 'x86', packageType: 'jdk' },
+      `https://api.azul.com/zulu/download/community/v1.0/bundles/?os=macos&ext=tar.gz&bundle_type=jdk&javafx=false&arch=x86&hw_bitness=32&release_status=ga`
+    ],
+    [
+      { version: '11', arch: 'x86', packageType: 'jre' },
+      `https://api.azul.com/zulu/download/community/v1.0/bundles/?os=macos&ext=tar.gz&bundle_type=jre&javafx=false&arch=x86&hw_bitness=32&release_status=ga`
+    ],
+    [
+      { version: '11-ea', arch: 'x86', packageType: 'jdk' },
+      `https://api.azul.com/zulu/download/community/v1.0/bundles/?os=macos&ext=tar.gz&bundle_type=jdk&javafx=false&arch=x86&hw_bitness=32&release_status=ea`
+    ],
```

Comment:
Let's reduce the number of test cases:
1) x64 + jdk
2) x86 + jdk
3) x64 + jre
4) x64 + jdk+fx
5) x64 + jre+fx

Since version is not used in url, it doesn't matter

Reply:
Done. Thank you.